jmix-flowui/flowui/src/main/java/io/jmix/flowui/facet/urlqueryparameters/DataGridFilterUrlQueryParametersBinder.java

```diff
@@ -134,8 +142,12 @@ protected void applyPropertyFilterParameters(List<String> params) {
             String propertyString = parameterString.substring(0, separatorIndex);
             String property = urlParamSerializer.deserialize(String.class,
                     filterUrlQueryParametersSupport.restoreSeparatorValue(propertyString));
+            MetaClass entityMetaClass = ((ContainerDataGridItems<?>) grid.getDataProvider()).getEntityMetaClass();
 
-            DataGridColumn<?> column = (DataGridColumn<?>) grid.getColumnByKey(property);
+            DataGridColumn<?> column = grid.getColumnByMetaPropertyPath(metadataTools.resolveMetaPropertyPath(
```

Comment:
Check case if we have several columns with different keys but bound to the same entity property